cpp/bench/prims/common/benchmark.hpp

```diff
@@ -113,8 +113,19 @@ class fixture {
   raft::device_resources handle;
   rmm::cuda_stream_view stream;
 
-  fixture() : stream{handle.get_stream()}
+  fixture(bool use_pool_memory_resource = false) : stream{handle.get_stream()}
   {
+    // Cache memory pool between test runs, since it is expensive to create.
+    // This speeds up the time required to run the select_k bench by over 3x.
+    // This is part of the fixture class here so that the pool will get cleaned
+    // up, rather than outliving the benchmarks that require it.
+    static std::unique_ptr<using_pool_memory_res> memory_pool;
+    if (use_pool_memory_resource) {
+      if (!memory_pool) { memory_pool.reset(new using_pool_memory_res()); }
+    } else if (memory_pool) {
+      memory_pool.reset();
```

Comment:
I'm not following what an empty `reset()` is doing here. I'm reading this line as "a new memory pool is created", but your analysis shows that's not the case. Could you please explain?

Reply:
This frees the memory being allocated by the memory pool - like in this case `use_pool_memory_resource` is false, and the empty reset will free up any previously allocated memory pool.